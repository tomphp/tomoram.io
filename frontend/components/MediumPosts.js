'use strict';

import React from 'react';
import MediumLogo from '../images/medium.png';

class MediumPost extends React.PureComponent {
  static defaultProps = {
    title: null,
  };

  render() {
    return (
      <div className="medium-posts__post">
        <h3 className="medium-posts__post-title">
          <a href={this.props.url} target="_blank">{this.props.title}</a>
        </h3>
        <div className="medium-posts__post-subtitle">{this.props.subtitle}</div>
      </div>
    );
  }
}

function renderPostItem(post, index) {
  return (
    <li key={`medium-post-${index}`}>
      <MediumPost
        title={post.title}
        url={`https://cloudnative.ly/${post.slug}-${post.id}`}
        subtitle={post.subtitle}
      />
    </li>
  );
}

export default class MediumPosts extends React.PureComponent {
  static defaultProps = {
    posts: [],
  };

  render() {
    return <div className="medium-posts">
      <h2 className="medium-posts__title">
        <img src={MediumLogo} alt="Medium Posts"/>
      </h2>
      <ul className="medium-posts__post-list">
        {this.props.posts.map(renderPostItem)}
      </ul>
    </div>;
  }
}
