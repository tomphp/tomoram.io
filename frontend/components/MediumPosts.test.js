'use strict';

import React from 'react';
import Enzyme from 'enzyme';
import MediumPosts from './MediumPosts';

describe('<MediumPosts/>', () => {
  const component = Enzyme.render(<MediumPosts posts={[]}/>);

  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('has block class of medium-posts', () => {
    expect(component.hasClass('medium-posts')).toBe(true);
  });
});
