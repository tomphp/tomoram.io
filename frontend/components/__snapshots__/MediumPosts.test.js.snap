// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediumPosts/> matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "medium-posts",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "medium-posts__title",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "Medium Posts",
              "src": "test-file-stub",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "h2",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "medium-posts__post-list",
          },
          "children": Array [],
          "name": "ul",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "medium-posts__post-list",
        },
        "children": Array [],
        "name": "ul",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "medium-posts__title",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "alt": "Medium Posts",
                "src": "test-file-stub",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "h2",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
